### ISSUE TITLE
Multiline admin field is broken

### ISSUE THREAD

**Issue by wintermute-84** [magento/magento2#8086]
### Preconditions

1. Magento version 2.1.3
2. PHP 7

### Steps to reproduce

1. Add field type="multiline" in any form in admin.
2. Save values in multiple fields

### Expected result

1. Values are displayed as saved
2. Ability to add new lines

### Actual result

<img width="751" alt="screen shot 2017-01-10 at 12 42 15" src="https://cloud.githubusercontent.com/assets/3521295/21799339/422a7518-d732-11e6-94be-4010f7129a0f.png">
1. The plus button (or any other way to add additional lines) is missing
Saved values are not exploded correctly when populating the fields.

**Comment by WaPoNe** [comment]
Any news on this issue?
I verified it on 2.1.7

**Comment by wintermute-84** [comment]
No, the functionality doesn't work.

BUT.

As a work around I've used other method:
See code bellow for a general idea.
![screen shot 2017-06-07 at 11 12 45](https://user-images.githubusercontent.com/3521295/26866625-fd613f56-4b72-11e7-80fb-d4a1f9e0ef9e.png)

system.xml
```
<field id="field_mapping" translate="label" sortOrder="410" showInDefault="1" showInWebsite="1" showInStore="1">
                    <label>Field Mapping</label>
                    <frontend_model>...\...\Block\Adminhtml\Form\Field\Mapping</frontend_model>
                    <backend_model>...\...\Model\Config\Source\Backend\Mapping</backend_model>
                </field>
```
Backend\Mapping
```
class Mapping extends \Magento\Framework\App\Config\Value
{
    protected $_customtabs = null;

    /**
     * @param \Magento\Framework\Model\Context $context
     * @param \Magento\Framework\Registry $registry
     * @param \Magento\Framework\App\Config\ScopeConfigInterface $config
     * @param \Magento\Framework\App\Cache\TypeListInterface $cacheTypeList
     * @param \Magento\Framework\Model\ResourceModel\AbstractResource $resource
     * @param \Magento\Framework\Data\Collection\AbstractDb $resourceCollection
     * @param array $data
     */
    public function __construct(
        \Magento\Framework\Model\Context $context,
        \Magento\Framework\Registry $registry,
        \Magento\Framework\App\Config\ScopeConfigInterface $config,
        \Magento\Framework\App\Cache\TypeListInterface $cacheTypeList,
        \Magento\Framework\Model\ResourceModel\AbstractResource $resource = null,
        \Magento\Framework\Data\Collection\AbstractDb $resourceCollection = null,
        array $data = []
    ) {
        parent::__construct($context, $registry, $config, $cacheTypeList, $resource, $resourceCollection, $data);
    }
    
    /**
     * Process data after load
     *
     * @return void
     */
    protected function _afterLoad()
    {
        $value = $this->getValue();
        $arr = unserialize($value);

        $this->setValue($arr);
    }

    /**
     * Prepare data before save
     *
     * @return void
     */
    public function beforeSave()
    {
        $value = $this->getValue();
        unset($value['__empty']);
        $arr = serialize($value);
        
        $this->setValue($arr);
    }
}
```
Field\Mapping
```
class Mapping extends \Magento\Config\Block\System\Config\Form\Field\FieldArray\AbstractFieldArray
{

    private $yesnoRenderer;

    /**
     * Prepare to render
     *
     * @return void
     */
    protected function _prepareToRender()
    {
        $this->addColumn('attribute_code', ['label' => __('Attribute Code')]);
        $this->addColumn('mapped_code', ['label' => __('Mapped Value')]);
        $this->addColumn('overwrite_status', ['label' => __('Overwrite?'), 'renderer'  => $this->getYesnoRenderer()]);
        $this->_addAfter = false;
        $this->_addButtonLabel = __('Add');
    }

    protected function getYesnoRenderer()
    {
        if (!$this->yesnoRenderer) {
            $this->yesnoRenderer = $this->getLayout()->createBlock(
                Yesno::class,
                '',
                ['data' => ['is_render_to_js_template' => true]]
            );
            $this->yesnoRenderer->setClass('overwrite_status');
        }
        return $this->yesnoRenderer;
    }

    /**
     * Prepare existing row data object
     *
     * @param \Magento\Framework\DataObject $row
     * @return void
     */
    protected function _prepareArrayRow(\Magento\Framework\DataObject $row)
    {
        $optionExtraAttr = [];

        $optionExtraAttr['option_' . $this->getYesnoRenderer()->calcOptionHash($row->getData('overwrite_status'))] =
            'selected="selected"';

        $row->setData('option_extra_attrs',$optionExtraAttr);
    }
}
```

**Comment by IlnitskiyArtem** [comment]
@Dukeofvampires, Thanks for reporting this issue. 
We've created internal ticket MAGETWO-69696 to address this issue.

**Comment by magento-engcom-team** [comment]
@wintermute-84, thank you for your report.
We've created internal ticket(s) MAGETWO-69696 to track progress on the issue.

**Comment by magento-engcom-team** [comment]
Hi @wintermute-84

This ticket has been marked as `"Triage Wanted"` due to low user involvement over time. Over the next 2 weeks we are looking for additional community feedback to decide if it should be archived or not. More information on this is available on the GitHub [wiki](https://github.com/magento/magento2/wiki/Triage-Wanted).

Thank you for collaboration.

**Comment by wintermute-84** [comment]
what about internal ticket MAGETWO-69696?

**Comment by magento-engcom-team** [comment]
@wintermute-84 Unfortunately internal ticket is still open.

**Comment by magento-engcom-team** [comment]
Hi @vivekkumarcedcoss. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @wintermute-84. Thank you for your report. 
The issue has been fixed in magento/magento2#20371 by @vivekkumarcedcoss in 2.3-develop branch
Related commit(s):
 - [ddb1137b96ff0ae93c2e78dff62e02da31f9a02a](https://github.com/magento/magento2/commit/ddb1137b96ff0ae93c2e78dff62e02da31f9a02a)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @wintermute-84. Thank you for your report. 
The issue has been fixed in magento/magento2#21561 by @vivekkumarcedcoss in 2.2-develop branch
Related commit(s):
 - [96c83295a26dd07e1819e310d2d31fef25fadec9](https://github.com/magento/magento2/commit/96c83295a26dd07e1819e310d2d31fef25fadec9)

The fix will be available with the upcoming 2.2.9 release.
